app/main.py

```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(file_path, errors)
+            for file_path, errors in linter_report.items()]
```

Comment:
Tente sempre deixar suas chaves, parênteses e colchetes em linhas diferentes das quais você implementa a lógica da sua list ou dict comprehension, principalmente se ela for muito grande, como é esse caso. Isso vai deixar seu código mais legível e também coerente com a forma que você estruturou o `return` das outras funções. 

```suggestion
    return [
        format_single_linter_file(file_path, errors)
        for file_path, errors in linter_report.items()
    ]
```